pkg/proto/configuration/filesystem/virtual/virtual.proto

```diff
@@ -139,6 +139,11 @@ message FUSEMountConfiguration {
   //
   // Recommended value: unset
   map<string, string> linux_backing_dev_info_tunables = 9;
+
+  // DirectMountStrict is like DirectMount but no fallback to fusermount is
+  // performed. If both DirectMount and DirectMountStrict are set,
+  // DirectMountStrict wins.
+  bool direct_mount_strict = 10;
```

Comment:
I'm always a big fan of making illegal states unrepresentable. There are only three behaviors:

- Always use fusermount
- Attempt to mount directly, falling back to fusermount
- Only attempt to mount directly

Yet two booleans give us four different options. Could you replace `direct_mount` and `direct_mount_strict` by an enum? Maybe something like this:

```proto
enum MountMethod {
  // Call into the fusermount utility to create the FUSE mount.
  FUSERMOUNT = 0;

  // Create the FUSE mount using the mount() system call. This can be
  // used in environments where the fusermount utility is not available,
  // such as the bb_worker container images.
  DIRECT = 1;

  // First attempt to create the FUSE mount using the mount() system
  // call. Upon failure, fall back to the fusermount utility.
  DIRECT_AND_FUSERMOUNT = 2;
}
```

Reply:
Updated!